lib/internal/bootstrap/loaders.js

```diff
@@ -208,6 +242,93 @@
+            this.reflect.exports[property].set(value);
+        };
+
+        const methodWrapMap = new WeakMap();
+        const methodUnwrapMap = new WeakMap();
+
+        const wrap = (target, name, value) => {
+          if (typeof value !== 'function' || !isNative(value)) {
+            return value;
+          }
+
+          if (methodWrapMap.has(value))
+            return methodWrapMap.get(value);
+
+          const p = new Proxy(value, {
+            apply: (t, thisArg, args) => {
+              if (thisArg === proxy || (this.reflect !== undefined &&
+                  this.reflect.namespace !== undefined &&
+                  thisArg === this.reflect.namespace)) {
+                thisArg = target;
+              }
+              return ReflectApply(t, thisArg, args);
+            },
+            __proto__: null,
+          });
+
+          methodWrapMap.set(value, p);
+          methodUnwrapMap.set(p, value);
+
+          return p;
+        };
+
+        const proxy = new Proxy(this.exports, {
+          set: (target, prop, value, receiver) => {
+            if (typeof value === 'function')
+              value = methodUnwrapMap.get(value) || value;
+            if (ReflectSet(target, prop, value, receiver)) {
+              update(prop, ReflectGet(target, prop, receiver));
+              return true;
+            }
+            return false;
+          },
+          defineProperty: (target, prop, descriptor) => {
+            if (ObjectHasOwnProperty.call(descriptor, 'value')) {
+              const { value } = descriptor;
+              if (typeof value === 'function')
+                descriptor.value = methodUnwrapMap.get(value) || value;
+            }
+            if (ReflectDefineProperty(target, prop, descriptor)) {
+              update(prop, ReflectGet(target, prop));
+              return true;
+            }
+            return false;
+          },
+          deleteProperty: (target, prop) => {
+            if (ReflectDeleteProperty(target, prop)) {
+              update(prop, undefined);
+              return true;
+            }
+            return false;
+          },
+          getOwnPropertyDescriptor: (target, prop) => {
+            const descriptor = ReflectGetOwnPropertyDescriptor(target, prop);
+            if (descriptor && ReflectHas(descriptor, 'value'))
+              descriptor.value = wrap(target, prop, descriptor.value);
+            return descriptor;
+          },
+          get: (target, prop, receiver) => {
+            const value = ReflectGet(target, prop, receiver);
+            if (prop === Symbol.toStringTag &&
+                typeof target !== 'function' &&
+                typeof value !== 'string') {
+              const toStringTag = ObjectToString.call(target).slice(8, -1);
+              return toStringTag === 'Object' ? value : toStringTag;
+            }
+            return wrap(target, prop, value);
+          },
+          __proto__: null,
```

Comment:
👆 It's a precaution since trap handlers, like descriptors, will read inherited traps/descriptors as well as own.
Does Node have a convention established for this?

Reply:
places needing to avoid `Object.prototype` use `Object.create` in the existing codebase but loader has some usage of `__proto__` in literals. No linting or style guide rules exist for Node to my knowledge on this topic.